Feat/integration specific ai builder

arifulhoque7 opened the pull request:
Close [issue](https://github.com/weDevsOfficial/wpuf-pro/issues/1180) , Close [1376](https://github.com/weDevsOfficial/wpuf-pro/issues/1376) , Related PRO [PR](https://github.com/weDevsOfficial/wpuf-pro/pull/1383)

<p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>AI Form Builder with Pre-built Integration Templates for WooCommerce, EDD, and The Events Calendar</strong></p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">📋 Overview</h3><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">This feature adds integration-specific AI form generation capabilities to the WP User Frontend AI Form Builder, enabling users to create specialized forms for popular WordPress plugins with pre-configured field structures and intelligent prompts.</p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">✨ Key Features</h3><h4 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">🎯<span> </span><strong>Integration Selector UI</strong></h4><ul style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>Added dropdown selector in the AI Form Builder input stage (post forms only)</li><li>Users can choose from:<ul style="padding-inline-start: 1.5em;"><li><strong>Regular Form</strong><span> </span>(default)</li><li><strong>WooCommerce Product Form</strong></li><li><strong>Easy Digital Downloads Product Form</strong></li><li><strong>The Events Calendar Event Form</strong></li></ul></li><li>Dynamic integration detection based on active plugins</li><li>Graceful fallback when integrations are not available</li></ul><h4 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">🤖<span> </span><strong>Specialized AI Prompts</strong></h4><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Added three comprehensive prompt templates with precise field structures:</p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><ol style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li><p style="display: inline;"><strong><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">wpuf-ai-prompt-woocommerce.md</code></strong><span> </span>(463 lines)</p><ul style="padding-inline-start: 1.5em;"><li>Complete WooCommerce product submission form structure</li><li>Includes: Product Name, Description, Pricing, Gallery, Catalog Visibility, Reviews</li><li>Sets<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">post_type: "product"</code><span> </span>automatically</li></ul></li><li><p style="display: inline;"><strong><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">wpuf-ai-prompt-edd.md</code></strong><span> </span>(325 lines)</p><ul style="padding-inline-start: 1.5em;"><li>Easy Digital Downloads product form template</li><li>Includes: Download Name, Description, Pricing, Files, Categories</li><li>Sets<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">post_type: "download"</code><span> </span>automatically</li></ul></li><li><p style="display: inline;"><strong><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">wpuf-ai-prompt-events-calendar.md</code></strong><span> </span>(408 lines)</p><ul style="padding-inline-start: 1.5em;"><li>The Events Calendar event submission form</li><li>Includes: Event Title, Description, Date/Time, Venue, Organizer</li><li>Sets<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">post_type: "tribe_events"</code><span> </span>automatically</li></ul></li></ol><h4 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">🔌<span> </span><strong>Backend API</strong></h4><ul style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>New REST endpoint:<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">GET /wpuf/v1/ai-form-builder/integrations</code></li><li>Returns available integrations based on active plugins</li><li>Integration detection logic in<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">AI_Manager.php</code></li><li>Integration-specific prompt loading in<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">FormGenerator.php</code></li></ul><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">📁 Modified Files</h3>
File | Changes | Lines
-- | -- | --
assets/js/components/AIFormBuilder.vue | Integration state management | +42
assets/js/components/FormInputStage.vue | Integration selector UI | +167
includes/AI/FormGenerator.php | Integration-specific prompts | +53
includes/AI/Form_Builder.php | Integration context passing | +15
includes/AI/RestController.php | Integrations endpoint | +86
includes/AI_Manager.php | Integration detection | +11

<p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>New Files:</strong></p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><ul style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">includes/AI/wpuf-ai-prompt-woocommerce.md</code><span> </span>(+463 lines)</li><li><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">includes/AI/wpuf-ai-prompt-edd.md</code><span> </span>(+325 lines)</li><li><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">includes/AI/wpuf-ai-prompt-events-calendar.md</code><span> </span>(+408 lines)</li></ul><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>Total: +1,548 lines, -22 lines</strong></p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">🎨 User Experience</h3><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>Before:</strong></p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><ul style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>Users had to manually describe WooCommerce/EDD/Event fields</li><li>No guidance on required WooCommerce meta fields</li><li>Generic AI prompts for all form types</li></ul><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>After:</strong></p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><ul style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>One-click selection of integration type</li><li>Pre-configured field structures matching plugin standards</li><li>Context-aware AI that understands WooCommerce<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">_regular_price</code>, EDD pricing tiers, Event date/time fields</li><li>Automatic<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">post_type</code><span> </span>configuration</li></ul><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">🌍 Multilingual Support</h3><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">All prompts include:</p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><ul style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li>Multi-language field label generation</li><li>Support for RTL languages (Arabic, Hebrew)</li><li>Language detection from user input</li><li>Automatic label translation when target language is specified</li></ul><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">🔧 Technical Details</h3><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>Integration Detection:</strong></p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><pre style="overflow-x: auto; max-width: 100%; white-space: pre; padding: 8px; border-radius: 4px; margin: 8px 0px; box-sizing: border-box; color: rgb(213, 206, 217); font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><code class="language-php" style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">// WooCommerce
class_exists('WooCommerce')

// Easy Digital Downloads
class_exists('Easy_Digital_Downloads')

// The Events Calendar
class_exists('Tribe__Events__Main')
</code></pre><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>API Response Format:</strong></p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><pre style="overflow-x: auto; max-width: 100%; white-space: pre; padding: 8px; border-radius: 4px; margin: 8px 0px; box-sizing: border-box; color: rgb(213, 206, 217); font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><code class="language-json" style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">{
  "success": true,
  "integrations": [
    {
      "id": "woocommerce",
      "label": "WooCommerce Product Form",
      "active": true
    }
  ]
}
</code></pre><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">✅ Testing Checklist</h3><ul class="contains-task-list" style="padding-inline-start: 1.5em; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span>Integration selector shows only when plugins are active</li><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span>AI generates correct field structure for each integration</li><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">post_type</code><span> </span>is set correctly in form settings</li><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span>WooCommerce meta fields use<span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">is_meta: "yes"</code></li><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span>Event dates use proper date/time pickers</li><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span>Multilingual labels work correctly</li><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span>Fallback to regular form when no integration selected</li><li class="task-list-item"><input type="checkbox" disabled="" checked="" style="appearance: none; margin: 2px 4px 2px 2px; width: 1em; height: 1em; border-color: rgb(69, 69, 69); border-style: solid; border-width: 1px; border-image: none 100% / 1 / 0 stretch; border-radius: 2px; position: relative; pointer-events: none; vertical-align: middle;"><span> </span>REST API endpoint returns correct integrations</li></ul><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">🚀 Ready for Review</h3><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">This feature significantly enhances the AI Form Builder by providing intelligent, plugin-specific form generation that reduces setup time and ensures compatibility with popular WordPress eCommerce and events plugins.</p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"><span> </span></span><p style="display: inline; color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong>Branch:</strong><span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">feat/integration-specific-ai-builder</code><br><strong>Base:</strong><span> </span><code style="font-family: monospace; color: rgb(215, 186, 125); background-color: rgba(255, 255, 255, 0.1); padding: 2px 4px; border-radius: 3px; font-size: 0.9em; word-break: break-word;">develop</code><br><strong>Commits:</strong><span> </span>1 feature commit<br><strong>Status:</strong><span> </span>✅ Ready for review</p><span style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;"></span><hr style="color: rgb(213, 206, 217); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, Roboto, sans-serif; font-size: 13px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(35, 38, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Added integration selection interface for creating platform-specific forms supporting WooCommerce, Events Calendar, Dokan, and other marketplace platforms
  * Integration-aware form templates with tailored generation prompts for each platform
  * Enhanced AI generation capabilities with improved token management across providers
  * Dynamic prompt templates based on selected integration

* **Style**
  * UI improvements to form generation interface
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

arifulhoque7 committed:
Feat: AI Form Builder Integrations for WooCommerce, EDD, and The Events Calendar

arifulhoque7 committed:
Merge branch 'weDevsOfficial:develop' into feat/integration-specific-ai-builder

arifulhoque7 committed:
Merge branch 'weDevsOfficial:develop' into feat/integration-specific-ai-builder

arifulhoque7 committed:
Improve AI form integration handling and prompt accuracy

Enhances integration persistence in the AI form builder, improves loading and selection UX for integrations, and updates prompt templates for WooCommerce, EDD, and Events Calendar to clarify required fields and template structure. Increases token limits for integration forms, adds detailed logging for AI requests and JSON errors, and ensures integration is included in API responses for regeneration. Also refines admin page detection and sanitization in AI_Manager.

arifulhoque7 commented:
@Rubaiyat-E-Mohammad vai, now check please

arifulhoque7 commented:
> Member

these are optional fields , the required fields for  woocommerce products are already attached with the form . @Rubaiyat-E-Mohammad vai

arifulhoque7 committed:
Update AI prompt for Woo Digital Product form

Expanded the AI instructions for WooCommerce Digital Product forms to include Downloadable Product and Downloadable Files fields.

arifulhoque7 committed:
Add digital product fields for WooCommerce Pro

Enhanced WooCommerce integration by adding support for digital product fields in Pro mode. Updated system prompts and documentation to include 'Downloadable Product' (radio_field) and 'Downloadable Files' (file_upload) as Pro-only fields, with clear instructions and JSON templates for their usage.

arifulhoque7 commented:
> The downloadable product field is a radio field with yes & no options. but showing a checkbox @arifulhoque7 vai <img alt="image" width="1097" height="276" src="https://private-user-images.githubusercontent.com/79570623/530697509-e993eace-57f7-4c4c-80cc-72f0280a6e66.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjcwMDk4ODksIm5iZiI6MTc2NzAwOTU4OSwicGF0aCI6Ii83OTU3MDYyMy81MzA2OTc1MDktZTk5M2VhY2UtNTdmNy00YzRjLTgwY2MtNzJmMDI4MGE2ZTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMjI5VDExNTk0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk2MDAxMmE1NjE1OGY0ZTZhMGFhMjZlNjZlNjNkYzU4ZTNlZDMyMGQwOGI0MDdkMmI4NDY5MzhkNWI5MjVmNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.pgLoh_DTt5mlrh1qjYSbqbvZC93V9jYuSjhYVWizi9o">

Now it should be fixed @Rubaiyat-E-Mohammad vai, check this

arifulhoque7 committed:
Set form_template based on post_type for integrations

Adds logic to assign the form_template field in form settings based on the post_type, ensuring proper integration with plugins like WooCommerce, EDD, and The Events Calendar. Introduces a private get_form_template_for_post_type() method to map post types to their corresponding form templates, allowing integrations to correctly trigger their hooks and save meta fields.

sapayth commented on includes/AI/FormGenerator.php:
```diff
@@ -599,6 +606,11 @@ private function generate_with_google($prompt, $options = []) {
         ];
 
+        // Log request details for debugging
+        error_log('WPUF AI: Google AI Request - Model: ' . $this->current_model);
```
remove the error and console logs

arifulhoque7 committed:
Merge branch 'weDevsOfficial:develop' into feat/integration-specific-ai-builder

arifulhoque7 committed:
Remove debug logging from AI form components and backend

Eliminated error_log debug statements from AIFormBuilder.vue, FormSuccessStage.vue, and FormGenerator.php to clean up code and prevent unnecessary logging in production. No functional changes were made.

arifulhoque7 commented on assets/js/components/AIFormBuilder.vue:
```diff
@@ -108,26 +112,34 @@ export default {
+                    session_id: this.getSessionId(),
+                    form_type: config.formType || 'post', // Pass form type to API
+                    provider: config.provider || 'openai'
+                    // Note: temperature and max_tokens are now handled by backend based on model configuration
```
done @sapayth vai

arifulhoque7 committed:
Merge branch 'weDevsOfficial:develop' into feat/integration-specific-ai-builder

arifulhoque7 committed:
Add vendor registration AI support for Dokan, WC Vendors, WCFM

Expanded AI form generation to support Dokan, WC Vendors, and WCFM vendor registration forms. Updated Vue components to show integration selector for registration/profile forms and load integration-specific prompt templates. Added new prompt files for each vendor integration. Enhanced backend to select integration-specific registration prompts and system instructions. Updated REST API to register new integrations and their requirements.

arifulhoque7 committed:
Improve token handling and settings merge in AI forms

Increased token limits for integration forms in FormGenerator and refactored token assignment logic for clarity. Updated Form_Builder to use array_replace_recursive for merging settings, ensuring nested defaults are preserved and preventing warnings with non-array input. Fixed WooCommerce integration check in RestController and ensured stale form_template values are cleared when no template is mapped. Standardized hook registration formatting in AI_Manager.

arifulhoque7 committed:
Fix prompt template rehydration and update AI prompt docs

Ensures selectedPrompt is correctly rehydrated from initialSelectedPrompt after promptTemplates are loaded in FormInputStage.vue. Also clarifies instructions in AI prompt documentation to specify 'Markdown' code blocks and improves formatting.

arifulhoque7 committed:
Fix token limit logic for GPT-5 and integration forms

Reorders the token parameter logic so that GPT-5 models always receive the higher token limit, even for integration forms. This ensures GPT-5's requirements are prioritized over integration-specific settings.

arifulhoque7 committed:
Add margin reset to step descriptions in FormProcessingStage

Added the 'wpuf-m-0' class to all step description paragraphs to ensure consistent margin reset and improved layout alignment in the form processing stage component.

sapayth commented on assets/js/components/FormInputStage.vue:
```diff
@@ -179,6 +213,188 @@ export default {
+                const restUrl = config.rest_url || (window.location.origin + '/wp-json/');
+                const nonce = config.nonce || '';
+
+                const response = await fetch(restUrl + 'wpuf/v1/ai-form-builder/integrations?form_type=' + this.formType, {
```
do not hardcode REST URL. use `wpufAIFormBuilder` object to pass the REST URL from PHP end

sapayth commented on includes/AI/RestController.php:
```diff
@@ -136,6 +143,22 @@ public function register_routes() {
         ]);
 
+        // Get available integrations endpoint
+        register_rest_route($this->namespace, '/' . $this->rest_base . '/integrations', [
```
follow our projects PHPCS guidelines for spacing

arifulhoque7 committed:
Refactor token limits and REST route registration

Centralized max token constants in Config.php and updated all usages in FormGenerator.php to reference these constants for consistency. Refactored RestController.php to use a more readable, array-based format for register_rest_route calls, improving maintainability. Added error handling for missing REST URL in FormInputStage.vue.

arifulhoque7 committed:
Refactor AI prompt handling and improve extensibility

Moves AI prompt files to a dedicated prompts directory and removes registration/integration-specific prompt files from the free plugin. Refactors FormInputStage.vue to load prompt templates and instructions from PHP-localized data, enabling easier extension by pro add-ons. Updates FormGenerator.php to use a filterable prompt file path and limits free plugin prompt support to basic, WooCommerce, and Events Calendar forms. Adds a filter in RestController.php for extensible AI integrations. Cleans up legacy prompt files and improves maintainability.

arifulhoque7 committed:
Refactor API endpoint usage to use config endpoints

Updated frontend components to use explicit API endpoint URLs from the config.endpoints object instead of constructing URLs from rest_url. Added error handling for missing endpoints. Updated Admin.php to provide these endpoint URLs to the frontend.

arifulhoque7 committed:
Reset invalid selected integration on update

Adds logic to clear the selected integration if it no longer exists in the updated list of available integrations, ensuring the UI stays in sync with backend data.

arifulhoque7 committed:
Merge remote-tracking branch 'origin/develop' into feat/integration-specific-ai-builder

arifulhoque7 committed:
Remove implicit defaults for form_type/provider

Stop falling back to hardcoded defaults for form_type and provider in AIFormBuilder.vue and FormSuccessStage.vue. The config values are now passed through directly (no 'post', 'openai', or 'google' fallbacks), so callers must provide explicit values if required. This makes provider/form_type handling consistent and avoids unintended defaults being sent to the backend.
